### PR TITLE
logind: process SIGTERM + SIGINT properly

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by yuwata** [comment]
Needs to be rebased.

**Comment by poettering** [comment]
force pushed a new rebased version. PTAL!

File: src/login/logind.c [hunk 1]
```diff
@@ -1223,13 +1224,13 @@ int main(int argc, char *argv[]) {
         mkdir_label("/run/systemd/users", 0755);
         mkdir_label("/run/systemd/sessions", 0755);
 
-        m = manager_new();
-        if (!m) {
-                r = log_oom();
+        r = manager_new(&m);
+        if (r < 0) {
+                log_error_errno(r, "Failed to allocate manager object: %m");
                 goto finish;
```
Comment (keszybz) [inline src/login/logind.c]: I don't think there's any need to notify `STOPPING=1` if `READY=1` was not send. So I'd either `return EXIT_FAILURE` or jump to the very end immediately, i.e. move the `finish` label down.

File: src/login/logind.c [hunk 2]
```diff
@@ -1221,14 +1221,12 @@ int main(int argc, char *argv[]) {
...
-         * watches in. Note that some applications might check for the
-         * existence of /run/systemd/seats/ to determine whether
-         * logind is available, so please always make sure this check
-         * stays in. */
-        mkdir_label("/run/systemd/seats", 0755);
-        mkdir_label("/run/systemd/users", 0755);
-        mkdir_label("/run/systemd/sessions", 0755);
+        /* Always create the directories people can create inotify watches in. Note that some applications might check
+         * for the existence of /run/systemd/seats/ to determine whether logind is available, so please always make
+         * sure this check stays in. */
```
Comment (keszybz) [inline src/login/logind.c]: "this check"?

**Comment by poettering** [comment]
Force pushed a new version with both points fixed, no other changes.

**Pull request merged**